app/uk/gov/hmrc/agentfirelationship/controllers/RelationshipController.scala

```diff
@@ -146,20 +150,54 @@ class RelationshipController @Inject() (
 
   private def setAuditData(arn: String, clientId: String, creds: Credentials)(implicit hc: HeaderCarrier): Future[AuditData] = {
     val auditData = new AuditData()
-    auditData.set("authProviderId", creds.providerId)
-    auditData.set("authProviderIdType", creds.providerType)
-    auditData.set("agentReferenceNumber", arn)
-    auditData.set("service", "personal-income-record")
-    auditData.set("clientId", clientId)
-    auditData.set("clientIdType", "ni")
-    Future successful auditData
+    if (creds.providerType == "GovernmentGateway") {
+      auditData.set("authProviderId", creds.providerId)
+      auditData.set("authProviderIdType", creds.providerType)
+      auditData.set("agentReferenceNumber", arn)
+      auditData.set("service", "personal-income-record")
+      auditData.set("clientId", clientId)
+      auditData.set("clientIdType", "ni")
+      Future successful auditData
+    } else if (creds.providerType == "PrivilegedApplication") {
+      auditData.set("authProviderId", creds.providerId)
+      auditData.set("authProviderIdType", creds.providerType)
+      auditData.set("agentReferenceNumber", arn)
+      auditData.set("service", "personal-income-record")
+      auditData.set("clientId", clientId)
+      Future successful auditData
+    } else throw new IllegalArgumentException("No providerType found in Credentials")
+
+  }
+
+  private def sendAuditEventForThisUser(credentials: Credentials, auditData: AuditData)(implicit hc: HeaderCarrier, request: Request[Any]): Future[Unit] = {
+    if (credentials.providerType == "GovernmentGateway")
+      auditService.sendTerminatedRelationshipEvent(auditData)
+    else if (credentials.providerType == "PrivilegedApplication")
+      auditService.sendHmrcLedDeleteRelationshipAuditEvent(auditData)
+    else throw new IllegalArgumentException("No providerType found in Credentials")
   }
 
-  private def forThisUser(requestedArn: Arn, requestedNino: Nino)(block: => Future[Result])(implicit taxIdentifier: TaxIdentifier) = {
+  private def forThisUser(requestedArn: Arn, requestedNino: Nino, strideRole: String)(block: => Future[Result])(implicit taxIdentifier: Option[TaxIdentifier]) = {
     taxIdentifier match {
-      case arn @ Arn(_) if requestedArn != arn => Future.successful(Forbidden)
-      case nino @ Nino(_) if requestedNino != nino => Future.successful(Forbidden)
-      case _ => block
+      case Some(t) => t match {
+        case arn @ Arn(_) if requestedArn != arn => {
+          Logger.warn("Arn does not match")
+          Future.successful(Forbidden)
+        }
+        case nino @ Nino(_) if requestedNino != nino => {
+          Logger.warn("Nino does not match")
+          Future.successful(Forbidden)
+        }
+        case _ => {
+          block
```

Comment:
I think `block` sounds like an opposite to what we actually do here, maybe name it `action`?